### PR TITLE
FAI-12219: Add delete session tracking to Faros destination

### PR DESCRIPTION
**Pull request by ted-faros**
## Description

Set session for reset delete queries.

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts [hunk 1]
```diff
@@ -406,20 +409,27 @@ export class GraphQLClient {
...
 
-  async deleteById(model: string, ids: string[]): Promise<void> {
+  async deleteById(
+    model: string,
+    ids: string[],
+    session: string
+  ): Promise<void> {
     const query = `mutation {
-      delete_${model}(where:{id: {_in:[
+      ctx: setCtx(args: {session: "${session}"}) {
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/common/graphql-client.ts:429]: is this a breaking change? Can we make it backward-compatible?

Comment (ted-faros) [reply destinations/airbyte-faros-destination/src/common/graphql-client.ts]: done.

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts [hunk 2]
```diff
@@ -363,8 +368,9 @@ export class GraphQLClient {
     await this.flush();
 
     const minRefreshedAt = new Date(this.resetLimitMillis).toISOString();
+    const deleteSessionId = randomUUID().toString();
     this.logger.info(
-      `Resetting data before ${minRefreshedAt} for origin ${origin}`
+      `Resetting data before ${minRefreshedAt} for origin ${origin} with session id ${deleteSessionId}`
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/common/graphql-client.ts:373]: everything, including logging should be conditioned by `this.supportsSetCtx`

**Review by tovbinm — APPROVED** [review]

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import {randomUUID} from 'node:crypto';
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/common/graphql-client.ts:1]: why not `import {randomUUID} from 'crypto'`?

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts (hunk 2 above)
Comment (ted-faros) [reply destinations/airbyte-faros-destination/src/common/graphql-client.ts]: done.

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts (hunk 3 above)
Comment (ted-faros) [reply destinations/airbyte-faros-destination/src/common/graphql-client.ts]: fixed

**Review by tovbinm — APPROVED** [review]

**Pull request merged**